internal/command/meta_providers.go

```diff
@@ -208,6 +208,9 @@ func (m *Meta) providerDevOverrideRuntimeWarnings() tfdiags.Diagnostics {
 		detailMsg.WriteString(fmt.Sprintf(" - %s in %s\n", addr.ForDisplay(), path))
 	}
 	detailMsg.WriteString("\nThe behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.")
+	if backendIsRemote {
+		detailMsg.WriteString("\n\nNote that provider development overrides have only been configured locally and the remote operation won't be affected by them")
+	}
```

Comment:
Just my two cents, but it feels aggressive to show the whole original warning to the user if we're going to then say "please disregard". We're making them read a lot. Consider swapping out the old message completely for a new one?

Reply:
Agreed 💯 . Just to expand on what Chris is suggesting here:
```
The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
```
This statement does not apply because like the following statement correctly suggests: 
```
...the remote operation won't be affected by them. 
```
The remote operation is unable to download the provider your overrides file points to and thus it can never apply changes that may cause state to be incompatible w/ published releases of the provider.  

That said... `backendIsRemote` is not a sufficient check here for to display either-or message. The reason being is even if I was using the remote/cloud backend... I can configure my workspace to use local execution. In which case the provider overrides changes that are applied _would_ affect state. You want to leverage:
```go
b, isRemoteBackend := be.(BackendWithRemoteTerraformVersion)
if isRemoteBackend && !b.IsLocalOperations() {
   // we are for-sure for-sure using remote execution
   // and we avoid potentially calling local execution mode a "remote operation"
}
```

Reply:
Makes sense 👍  @sebasslash had not thought of that edge case, good eyes!